### ISSUE TITLE
Unable to compile pad/unpad from Flash Attention 2

### ISSUE THREAD

**Issue by conceptofmind** [pytorch/pytorch#148845]
### 🐛 Describe the bug

Hello all,

I am attempting to compile a model that is unpadding and padding the input ids for an encoder with Flash Attention 2. The pad and unpad code can be found here: https://github.com/Dao-AILab/flash-attention/blob/main/flash_attn/bert_padding.py#L98

Example code:
```python
        x, indices, cu_seqlens, max_seqlen, _ = unpad_input(
            inputs=x, attention_mask=attn_mask
        )
        x = self.embed(x)
        x = self.transformer(x, attn_mask, cu_seqlens, max_seqlen)
        x = self.lmhead(x)
```

### Error logs

When trying to compile the model I receive this graph break:
```
W0308 14:43:37.982000 21720 site-packages/torch/_dynamo/variables/tensor.py:869] [7/0] Graph break from `Tensor.item()`, consider setting:
W0308 14:43:37.982000 21720 site-packages/torch/_dynamo/variables/tensor.py:869] [7/0]     torch._dynamo.config.capture_scalar_outputs = True
W0308 14:43:37.982000 21720 site-packages/torch/_dynamo/variables/tensor.py:869] [7/0] or:
W0308 14:43:37.982000 21720 site-packages/torch/_dynamo/variables/tensor.py:869] [7/0]     env TORCHDYNAMO_CAPTURE_SCALAR_OUTPUTS=1
W0308 14:43:37.982000 21720 site-packages/torch/_dynamo/variables/tensor.py:869] [7/0] to include these operations in the captured graph.
W0308 14:43:37.982000 21720 site-packages/torch/_dynamo/variables/tensor.py:869] [7/0] 
W0308 14:43:37.982000 21720 site-packages/torch/_dynamo/variables/tensor.py:869] [7/0] Graph break: from user code at:
W0308 14:43:37.982000 21720 site-packages/torch/_dynamo/variables/tensor.py:869] [7/0]   File "/models/attn.py", line 171, in torch_dynamo_resume_in__upad_input_at_170
W0308 14:43:37.982000 21720 site-packages/torch/_dynamo/variables/tensor.py:869] [7/0]     max_seqlen_in_batch = seqlens_in_batch.max().item()
W0308 14:43:37.982000 21720 site-packages/torch/_dynamo/variables/tensor.py:869] [7/0] 
W0308 14:43:37.982000 21720 site-packages/torch/_dynamo/variables/tensor.py:869] [7/0] 
tensor([[-1.0541, -0.1679,  0.2720,  ..., -0.0505,  0.5067, -0.0918],
        [-0.8398, -0.2256,  0.5115,  ..., -0.2662,  0.0860, -0.1525],
        [-0.2781, -0.3775,  0.3996,  ..., -0.1714,  0.7148,  0.3041]],
       device='cuda:0', grad_fn=<CompiledFunctionBackward>)
Number of parameters in torch model: 34554170
```
https://github.com/Dao-AILab/flash-attention/blob/5639b9d26dac63d912d6815cb4369250f6cef764/flash_attn/bert_padding.py#L115

### Versions

python3 collect_env.py
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 24353  100 24353    0     0  68213      0 --:--:-- --:--:-- --:--:-- 68407
Collecting environment information...
PyTorch version: 2.6.0+cu126
Is debug build: False
CUDA used to build PyTorch: 12.6
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (x86_64)
GCC version: (Ubuntu 12.3.0-1ubuntu1~22.04) 12.3.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.10.14 | packaged by conda-forge | (main, Mar 20 2024, 12:45:18) [GCC 12.3.0] (64-bit runtime)
Python platform: Linux-5.15.0-134-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 12.6.85
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3090
Nvidia driver version: 560.35.05
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        39 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               12
On-line CPU(s) list:                  0-11
Vendor ID:                            GenuineIntel
Model name:                           Intel(R) Core(TM) i5-10600K CPU @ 4.10GHz
CPU family:                           6
Model:                                165
Thread(s) per core:                   2
Core(s) per socket:                   6
Socket(s):                            1
Stepping:                             5
CPU max MHz:                          4800.0000
CPU min MHz:                          800.0000
BogoMIPS:                             8199.79
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault invpcid_single ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx rdseed adx smap clflushopt intel_pt xsaveopt xsavec xgetbv1 xsaves dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp pku ospke md_clear flush_l1d arch_capabilities
Virtualization:                       VT-x
L1d cache:                            192 KiB (6 instances)
L1i cache:                            192 KiB (6 instances)
L2 cache:                             1.5 MiB (6 instances)
L3 cache:                             12 MiB (1 instance)
NUMA node(s):                         1
NUMA node0 CPU(s):                    0-11
Vulnerability Gather data sampling:   Mitigation; Microcode
Vulnerability Itlb multihit:          KVM: Mitigation: VMX disabled
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Mitigation; Enhanced IBRS
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop
Vulnerability Srbds:                  Mitigation; Microcode
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] nvidia-cublas-cu11==11.11.3.6
[pip3] nvidia-cublas-cu12==12.6.4.1
[pip3] nvidia-cuda-cupti-cu11==11.8.87
[pip3] nvidia-cuda-cupti-cu12==12.6.80
[pip3] nvidia-cuda-nvrtc-cu11==11.8.89
[pip3] nvidia-cuda-nvrtc-cu12==12.6.77
[pip3] nvidia-cuda-runtime-cu11==11.8.89
[pip3] nvidia-cuda-runtime-cu12==12.6.77
[pip3] nvidia-cudnn-cu11==8.7.0.84
[pip3] nvidia-cudnn-cu12==9.5.1.17
[pip3] nvidia-cufft-cu11==10.9.0.58
[pip3] nvidia-cufft-cu12==11.3.0.4
[pip3] nvidia-curand-cu11==10.3.0.86
[pip3] nvidia-curand-cu12==10.3.7.77
[pip3] nvidia-cusolver-cu11==11.4.1.48
[pip3] nvidia-cusolver-cu12==11.7.1.2
[pip3] nvidia-cusparse-cu11==11.7.5.86
[pip3] nvidia-cusparse-cu12==12.5.4.2
[pip3] nvidia-cusparselt-cu12==0.6.3
[pip3] nvidia-nccl-cu11==2.20.5
[pip3] nvidia-nccl-cu12==2.21.5
[pip3] nvidia-nvjitlink-cu12==12.6.85
[pip3] nvidia-nvtx-cu11==11.8.86
[pip3] nvidia-nvtx-cu12==12.6.77
[pip3] pytorch-triton==3.0.0+45fff310c8
[pip3] rotary-embedding-torch==0.5.3
[pip3] torch==2.6.0+cu126
[pip3] torchaudio==2.6.0+cu126
[pip3] torchvision==0.21.0+cu126
[pip3] triton==3.2.0
[conda] numpy                     1.26.4                   pypi_0    pypi
[conda] nvidia-cublas-cu11        11.11.3.6                pypi_0    pypi
[conda] nvidia-cublas-cu12        12.6.4.1                 pypi_0    pypi
[conda] nvidia-cuda-cupti-cu11    11.8.87                  pypi_0    pypi
[conda] nvidia-cuda-cupti-cu12    12.6.80                  pypi_0    pypi
[conda] nvidia-cuda-nvrtc-cu11    11.8.89                  pypi_0    pypi
[conda] nvidia-cuda-nvrtc-cu12    12.6.77                  pypi_0    pypi
[conda] nvidia-cuda-runtime-cu11  11.8.89                  pypi_0    pypi
[conda] nvidia-cuda-runtime-cu12  12.6.77                  pypi_0    pypi
[conda] nvidia-cudnn-cu11         8.7.0.84                 pypi_0    pypi
[conda] nvidia-cudnn-cu12         9.5.1.17                 pypi_0    pypi
[conda] nvidia-cufft-cu11         10.9.0.58                pypi_0    pypi
[conda] nvidia-cufft-cu12         11.3.0.4                 pypi_0    pypi
[conda] nvidia-curand-cu11        10.3.0.86                pypi_0    pypi
[conda] nvidia-curand-cu12        10.3.7.77                pypi_0    pypi
[conda] nvidia-cusolver-cu11      11.4.1.48                pypi_0    pypi
[conda] nvidia-cusolver-cu12      11.7.1.2                 pypi_0    pypi
[conda] nvidia-cusparse-cu11      11.7.5.86                pypi_0    pypi
[conda] nvidia-cusparse-cu12      12.5.4.2                 pypi_0    pypi
[conda] nvidia-cusparselt-cu12    0.6.3                    pypi_0    pypi
[conda] nvidia-nccl-cu11          2.20.5                   pypi_0    pypi
[conda] nvidia-nccl-cu12          2.21.5                   pypi_0    pypi
[conda] nvidia-nvjitlink-cu12     12.6.85                  pypi_0    pypi
[conda] nvidia-nvtx-cu11          11.8.86                  pypi_0    pypi
[conda] nvidia-nvtx-cu12          12.6.77                  pypi_0    pypi
[conda] pytorch-triton            3.0.0+45fff310c8          pypi_0    pypi
[conda] rotary-embedding-torch    0.5.3                    pypi_0    pypi
[conda] torch                     2.6.0+cu126              pypi_0    pypi
[conda] torchaudio                2.6.0+cu126              pypi_0    pypi
[conda] torchvision               0.21.0+cu126             pypi_0    pypi
[conda] triton                    3.2.0                    pypi_0    pypi

cc @chauhang @penguinwu @zou3519 @ydwu4 @bdhirsh @Chillee @drisspg @yanboliang @BoyuanFeng

**Comment by IvanKobzarev** [comment]
@conceptofmind 
By default graphs with Tensor.item() could not be compiled.
It needs explicit set `torch._dynamo.config.capture_scalar_outputs = True` before compilation.
